### PR TITLE
[RPW]Fix docs from core package

### PR DESCRIPTION
**Pull request by alex-vazquez**
### Purpose of this PR
Adding API docs to Core package.

### CODE REVIEW

File: com.unity.render-pipelines.core/Editor/Lighting/LightUnit/LightUnitSlider.cs [hunk 1]
```diff
@@ -22,25 +25,48 @@ static class SliderConfig
...
+        /// <param name="descriptor">The <see cref="LightUnitSliderUIDescriptor"/></param>
         public LightUnitSlider(LightUnitSliderUIDescriptor descriptor)
         {
             m_Descriptor = descriptor;
         }
 
+        /// <summary>
+        /// Modifies the <see cref="SerializedObject"/> for this Light slider
+        /// </summary>
+        /// <param name="serialized"></param>
```
Comment (arttu-peltonen) [inline com.unity.render-pipelines.core/Editor/Lighting/LightUnit/LightUnitSlider.cs:58]: empty content here?

File: com.unity.render-pipelines.core/Editor/MaterialUpgrader.cs [hunk 2]
```diff
@@ -432,9 +433,11 @@ public static void UpgradeProjectFolder(List<MaterialUpgrader> upgraders, HashSe
         /// <param name="flags">Material Upgrader flags.</param>
         public static void Upgrade(Material material, MaterialUpgrader upgrader, UpgradeFlags flags)
         {
-            var upgraders = new List<MaterialUpgrader>();
-            upgraders.Add(upgrader);
-            Upgrade(material, upgraders, flags);
+            using (ListPool<MaterialUpgrader>.Get(out List<MaterialUpgrader> upgraders))
```
Comment (arttu-peltonen) [inline com.unity.render-pipelines.core/Editor/MaterialUpgrader.cs:436]: What's this about?

File: com.unity.render-pipelines.core/Runtime/Common/SerializedDictionary.cs [hunk 3]
```diff
@@ -3,27 +3,54 @@
...
-    //
     /// <summary>
-    /// Serialized Dictionary
+    /// Unity can't serialize Dictionary so here's a custom wrapper that does. Note that you have to
+    /// extend it before it can be serialized as Unity won't serialized generic-based types either.
     /// </summary>
-    /// <typeparam name="K">Key Type</typeparam>
-    /// <typeparam name="V">Value Type</typeparam>
+    /// <typeparam name="K">The key type</typeparam>
+    /// <typeparam name="V">The value</typeparam>
```
Comment (arttu-peltonen) [inline com.unity.render-pipelines.core/Runtime/Common/SerializedDictionary.cs:11]: value -> value type

File: com.unity.render-pipelines.core/Runtime/Debugging/Prefabs/Scripts/DebugUIHandlerMessageBox.cs [hunk 4]
```diff
@@ -36,6 +36,12 @@ internal override void SetWidget(DebugUI.Widget widget)
             }
         }
 
+        /// <summary>
+        /// Method called when the box is selected
+        /// </summary>
+        /// <param name="fromNext">If is from next</param>
+        /// <param name="previous">The previous <see cref="DebugUIHandlerWidget"/></param>
+        /// <returns></returns>
```
Comment (arttu-peltonen) [inline com.unity.render-pipelines.core/Runtime/Debugging/Prefabs/Scripts/DebugUIHandlerMessageBox.cs:44]:
```suggestion
        /// <summary>
        /// OnSelection implementation.
        /// </summary>
        /// <param name="fromNext">True if the selection wrapped around.</param>
        /// <param name="previous">Previous widget.</param>
        /// <returns>True if the selection is allowed.</returns>
```

File: com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs [hunk 5]
```diff
@@ -233,42 +233,40 @@ public override bool AllocateTexture(CommandBuffer cmd, ref Vector4 scaleOffset,
...
         /// </summary>
-        /// <param name="texture">Source texture.</param>
-        /// <returns>True on success, false otherwise.</returns>
+        /// <param name="texture">The source texture</param>
+        /// <returns>True if the space is reserved</returns>
         public bool ReserveSpace(Texture texture) => ReserveSpace(texture, texture.width, texture.height);
 
         /// <summary>
-        /// Reserve space from atlas for a texture.
+        /// Reserves the space on the texture atlas
```
Comment (arttu-peltonen) [inline com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs:243]: "Reserves the space on the texture atlas" is not grammatically correct IMO, it should be "Reserves space from the texture atlas"

File: com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs [hunk 6]
```diff
@@ -233,42 +233,40 @@ public override bool AllocateTexture(CommandBuffer cmd, ref Vector4 scaleOffset,
...
         /// </summary>
-        /// <param name="texture">Source texture.</param>
-        /// <param name="width">Request width in pixels.</param>
-        /// <param name="height">Request height in pixels.</param>
-        /// <returns>True on success, false otherwise.</returns>
+        /// <param name="texture">The source texture</param>
+        /// <param name="width">The width</param>
+        /// <param name="height">The height</param>
```
Comment (arttu-peltonen) [inline com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs:247]: Why remove "in pixels" here, I think it's better to keep it.

File: com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs [hunk 7]
```diff
@@ -233,42 +233,40 @@ public override bool AllocateTexture(CommandBuffer cmd, ref Vector4 scaleOffset,
...
+        /// <returns>True if the space is reserved</returns>
```
Comment (arttu-peltonen) [inline com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs:248]: I would prefer the original wording here

File: com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs [hunk 8]
```diff
@@ -233,42 +233,40 @@ public override bool AllocateTexture(CommandBuffer cmd, ref Vector4 scaleOffset,
...
         public bool ReserveSpace(Texture texture, int width, int height)
             => ReserveSpace(GetTextureID(texture), width, height);
 
-
         /// <summary>
-        /// Reserve space from atlas for a texture.
-        /// Pass width and height for CubeMap (use 2*width) & Texture2D (use width).
+        /// Reserves the space on the texture atlas
```
Comment (arttu-peltonen) [inline com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs:253]: IMO should be "Reserves space from the texture atlas".

Also seems like the removed extra info could be useful to someone so why not keep it?

File: com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs [hunk 9]
```diff
@@ -233,42 +233,40 @@ public override bool AllocateTexture(CommandBuffer cmd, ref Vector4 scaleOffset,
...
+        /// <param name="textureB">The source texture B</param>
+        /// <param name="width">The width</param>
+        /// <param name="height">The height</param>
+        /// <returns>True if the space is reserved</returns>
         public bool ReserveSpace(Texture textureA, Texture textureB, int width, int height)
             => ReserveSpace(GetTextureID(textureA, textureB), width, height);
 
         /// <summary>
-        /// Reserve space from atlas for a texture.
+        /// Reserves the space on the texture atlas
```
Comment (arttu-peltonen) [inline com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs:264]: "Reserves space from the texture atlas"

**Review by arttu-peltonen — CHANGES_REQUESTED** [review]
Looks great, thanks for doing the work! Minor comments on some edits.

File: com.unity.render-pipelines.core/Editor/MaterialUpgrader.cs (hunk 2 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Editor/MaterialUpgrader.cs:436]: This is an list pool, that we are not generating garbage with the new List<MaterialUpgrader>, as we are calling each material and creating a new List, I saw it while updating the docs.

Comment (arttu-peltonen) [reply com.unity.render-pipelines.core/Editor/MaterialUpgrader.cs:436]: Neat, I should look into that. Didn't know this type of thing existed.

File: com.unity.render-pipelines.core/Editor/Lighting/LightUnit/LightUnitSlider.cs (hunk 1 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Editor/Lighting/LightUnit/LightUnitSlider.cs:58]:
```suggestion
        /// <param name="serialized"></param>
```
```suggestion
        /// <param name="serialized">The serialized object</param>
```

File: com.unity.render-pipelines.core/Runtime/Common/SerializedDictionary.cs (hunk 3 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Runtime/Common/SerializedDictionary.cs:11]:
```suggestion
    /// <typeparam name="V">The value type</typeparam>
```

File: com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs (hunk 5 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs:243]:
```suggestion
        /// Reserves space from the texture atlas
```

File: com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs (hunk 6 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs:247]:
```suggestion
        /// <param name="width">The width in pixels</param>
        /// <param name="height">The height in pixels</param>
```

File: com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs (hunk 8 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs:253]:
```suggestion
        /// Reserves space from the texture atlas
```

File: com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs (hunk 9 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs:264]:
```suggestion
        /// Reserves space from the texture atlas
```

File: com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs (hunk 7 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Runtime/Textures/PowerOfTwoTextureAtlas.cs:248]: I think is obvious the original wording, that's why I changed.

**Review by arttu-peltonen — APPROVED** [review]

File: com.unity.render-pipelines.core/Runtime/Common/SerializedDictionary.cs [hunk 10]
```diff
@@ -3,27 +3,54 @@
...
+        /// <param name="val">The value</param>
+        /// <returns>The value</returns>
         public override V DeserializeValue(V val) => val;
     }
 
+    /// <summary>
+    /// Dictionary that can serialize keys and values as other types
+    /// </summary>
+    /// <typeparam name="K">The key</typeparam>
+    /// <typeparam name="V"></typeparam>
```
Comment (RSlysz) [inline com.unity.render-pipelines.core/Runtime/Common/SerializedDictionary.cs:51]:
```suggestion
    /// <typeparam name="K">The key type</typeparam>
    /// <typeparam name="V">The value type</typeparam>
```

**Review by RSlysz — APPROVED** [review]

**Review by theo-pnv — APPROVED** [review]

**Pull request merged**